### PR TITLE
refactor: run full validation for aks-engine deploy

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

This is a simpler method of running cluster configuration validation during `aks-engine deploy`

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
related to https://github.com/Azure/aks-engine/issues/1281

**Comment by CecileRobertMichon** [comment]
this PR conflicts with https://github.com/Azure/aks-engine/pull/1228/files which moved validation to a common function between deploy and generate

File: cmd/deploy.go [hunk 1]
```diff
@@ -80,8 +79,10 @@ func newDeployCmd() *cobra.Command {
 			if err := dc.loadAPIModel(); err != nil {
 				return errors.Wrap(err, "loading API model")
 			}
-			if _, _, err := dc.validateApimodel(); err != nil {
-				return errors.Wrap(err, "validating API model after populating values")
+			if dc.apiVersion == "vlabs" {
```
Comment (CecileRobertMichon) [inline cmd/deploy.go]: if not vlabs, maybe we should warn the user that validation was skipped?

File: cmd/deploy.go [hunk 2]
```diff
@@ -354,28 +355,9 @@ func autofillApimodel(dc *deployCmd) error {
...
 
-func (dc *deployCmd) validateApimodel() (*api.ContainerService, string, error) {
-	apiloader := &api.Apiloader{
-		Translator: &i18n.Translator{
-			Locale: dc.locale,
-		},
-	}
-
-	p := dc.containerService.Properties
-	if strings.ToLower(p.OrchestratorProfile.OrchestratorType) == "kubernetes" {
```
Comment (jackfrancis) [inline cmd/deploy.go]: For the life of me I can't figure out how to reproduce a deploy command matching this validation failure that actually gets this far in the flow. Basically, I don't know how one would use MSI + `aks-engine deploy`

Comment (CecileRobertMichon) [reply cmd/deploy.go]: it's very possible that it's outdated, this is really old code

**Pull request merged**